File: CreationalPattern/FactoryPattern/Java/Plants/WallNut.java

package Plants;

import Interface.Plant;

public class WallNut implements Plant {
    // Var parameters
    private String name;
    private int health;
    private int price;
    // TODO parameters

    // Constructor
    public WallNut() {
        super();
        this.name = "WallNut";
        this.health = 200;
        this.price = 50;
    }

    public WallNut(String name, int health, int price) {
        super();
        this.name = name;
        this.health = health;
        this.price = price;
    }

    // Override method show
    @Override
    public void show() {
        System.out.println("Name: " + name);
        System.out.println("Health: " + health);
        System.out.println("Price: " + price);
    }

    // Getter and Setter 
    // TODO add ...
}

File: CreationalPattern/FactoryPattern/Golang/Plants/WallNut.go

package Plants

/**
 * WallNut struct, implement Plant interface
 */
type WallNut struct {
	Name  string
	Price int
	// TODO add more attribute ...
}

// Constructor for WallNut
func NewWallNut(params...interface{}) *WallNut {
	p := &WallNut {
		Name: "WallNut",
		Price: 50,
	}

	len := len(params)

	if (len > 0) {
		p.Name = params[0].(string)
	}

	if (len > 1) {
		p.Price = params[1].(int)
	}

	return p
}

func (s *WallNut) PlantName() string {
	return s.Name
}

func (s *WallNut) PlantPrice() int {
	return s.Price
}